server/src/module/student/student.service.ts

```diff
@@ -102,16 +110,69 @@ export class StudentService implements CRUDService<IStudent, StudentDTO, Student
      *
      * @throws `NotFoundException` - If the tutorial or the team of the student could not be found.
      */
-    async create(dto: StudentDTO): Promise<IStudent> {
-        const { tutorial: tutorialId, team: teamId } = dto;
+    async create(dto: CreateStudentDTO): Promise<IStudent> {
+        const { team: teamId, tutorial: tutorialId } = dto;
         const tutorial = await this.tutorialService.findById(tutorialId);
         const team = !!teamId
             ? await this.teamService.findById({
                   tutorialId: tutorial.id,
                   teamId,
               })
             : undefined;
+        const student = await this.createStudent(dto, tutorial, team);
+        await this.repository.flush();
+        return student.toDTO();
+    }
+
+    /**
+     * Creates many students from the given DTO and returns the created students.
+     * For each student, if team is provided, if the team is the number of an existing team, the student is put into that team,
+     * otherwise a new Team is created and associated with the provided string.
+     *
+     * @param dto DTO with the information for the students to create.
+     *
+     * @returns Created students.
+     *
+     * @throws `NotFoundException` - If the tutorial could not be found.
+     */
+    async createMany(dto: CreateStudentsDTO): Promise<IStudent[]> {
+        const tutorial = await this.tutorialService.findById(dto.tutorial);
+        const teamNos = new Set(dto.students.map((student) => student.team));
+        const allTeams = await this.teamService.findAllTeamsInTutorial(dto.tutorial);
+        const teamLookup = new Map<string | undefined, Team>();
+        teamNos.forEach((teamNo) => {
+            if (teamNo && /^[0-9]+$/.test(teamNo)) {
+                const team = allTeams.find((it) => it.teamNo === parseInt(teamNo));
+                if (team) {
+                    teamLookup.set(teamNo, team);
+                }
+            }
+        });
+        teamNos.forEach((teamNo) => {
+            if (teamNo && !teamLookup.has(teamNo)) {
+                teamLookup.set(teamNo, this.teamService.createTeamWithoutStudents(tutorial));
+            }
+        });
+
+        const students = await Promise.all(
+            dto.students.map((student) => {
+                return this.createStudent(student, tutorial, teamLookup.get(student.team));
+            })
+        );
+        await this.repository.flush();
+        return students.map((student) => student.toDTO());
+    }
 
+    /**
+     * Creates a student from the given DTO and returns the created student.
+     *
+     * @param dto DTO with the information for the student to create.
+     * @param tutorial the Tutorial the created Student is created at.
+     * @param team the Team the created Student is put in.
+     *
+     * @returns Created student.
+     */
+    async createStudent(dto: StudentDTO, tutorial: Tutorial, team?: Team): Promise<Student> {
```

Comment:
Can `createStundet` be private?

Reply:
yes, fixed with 2c646882758f718ea9ccc8bac7c277152b096cf2